### ISSUE TITLE
Create documentation for building on a local machine

### ISSUE THREAD

**Issue by sinistersnare** [rust-lang/rust-playpen#86]
Hey, I am on a windows machine on my laptop, and would like to contribute, I have been surveying the source code for dependencies and I think i will be able to publish a list of requirements.This will include the contents of `pip freeze`, and perhaps a vagrant install script? This way there can be reproducible deploys, which would be super easy for local testing.

Ideas?

**Comment by svanellewee** [comment]
@sinistersnare have you had any progress on this?

I've only just started looking at this repo, but I don't think cross platform functionality has been considered yet.  

Check this diff, it's not really PR worthy material but might help indicate where we can start?

Problems I see are 
- the dash scripts for calling rust.. I've made mine bash, you would need a batch equivalent? Perhaps we can just make them python as well to make it properly cross platform?
- playpen dependency seems to require a whole lot of work to get going locally, perhaps we can deactivate it if this is not running on a "production" environment.. though how would we know when we're on production?

I've made a diff of what I tried to do to make it work on my BSD based system..

https://github.com/svanellewee/rust-playpen/compare/f886864dc754b52c0284c537b4aef6e499b59ab6...svanellewee:master

**Comment by sinistersnare** [comment]
Ive actually completely forgotten about this :) School was really busy right after I wrote this issue.

If you want to take the reigns on this please do, I think having clear cut dependencies and venv setup would be amazing. 

Thanks so much for your work! Sorry that I am so lazy :p

**Comment by edunham** [comment]
@sinistersnare, @svanellewee  Welcome! It's great to start getting other contributors to the infrastructure involved, and I'm sure you'll find all kinds of previously-unnoticed bugs (especially missing documentation) by building a copy of it yorself! 

The possibly bad news is that it's not possible to build rust-playpen on any platform other than Linux with systemd right now, because of the underlying [Playpen][https://github.com/thestinger/playpen] sandboxing tool's requirements. It actually only builds on Arch at the moment, due to some assumptions in `init.sh` about the availability of Arch-specific scripts, but I'm working on making it run on Ubuntu in order to get the special Arch snowflake out of Rust's production infrastructure. I'll also document the dependencies as I go along -- the end result will be an Ansible playbook that you can point at an Ubuntu box, run, and end up with a working rust-playpen installation.

@sinistersnare, I think the best bet for making progress on this right now would be to spin up an Arch VM by following instructions like [these][http://www.cs.columbia.edu/~jae/4118/arch-setup-2015-1.html].

**Comment by sinistersnare** [comment]
Yeah I figured it would only run on Arch, thats what the creator uses. I made the issue so that we can be aware that we should document things like that. 

I tihnk we need a couple things so we can have reproducible builds:

1) Vagrant usage
2) requirements.txt
3) a useful section in the README.

I am fine with only being able to run on Arch, but the fact that it was not documented is a problem.

**Comment by edunham** [comment]
Documented the IRC bot setup in #132. Haven't documented webserver or `init.sh` stuff yet.
